Libs/RaptureAPI/src/main/java/rapture/common/client/BaseHttpApi.java

```diff
@@ -250,6 +251,16 @@ public long getContentLength() {
     }
 
     protected GeneralResponse responseFromJson(String responseObjectJson) {
+        // if the servlet throws an exception - which happens if a Reflex assertion fails - then extract the info
+        if (responseObjectJson.startsWith("<!DOCTYPE html>")) {
+            log.warn(responseObjectJson);
+            int rootCause = responseObjectJson.indexOf("root cause");
+            if (rootCause > 0) {
+                int skipPre = responseObjectJson.indexOf("<pre>", rootCause);
+                String cause = responseObjectJson.substring(skipPre + 5, responseObjectJson.indexOf("</pre>", skipPre)).replaceAll("&quot;", "\"");
+                throw new RaptureException("", 500, cause);
```

Comment:
500 seems the most suitable for the general case. I wrote this specifically to catch the case where a Reflex test case assertion fails and to extract the error message. In such a case we currently get a 500 (which is embedded in the text). I considered extracting it in the same way.

Another approach might be to check the cause in more detail return a specific HTTP code to cover the case of an assertion fail and save 500 for genuine internal server errors. I'm tempted to use 418 or an unused 5xx code.